modules/tinymce/src/core/main/ts/api/html/Schema.ts

```diff
@@ -571,7 +571,7 @@ const Schema = (settings: SchemaSettings = {}): Schema => {
 
   const isBlock = (name: string): boolean => Obj.has(getBlockElements(), name);
 
-  const isInline = (name: string): boolean => isValid(name) && !isBlock(name);
+  const isInline = (name: string): boolean => !Strings.startsWith(name, '#') && isValid(name) && !isBlock(name);
```

Comment:
I feel like we should add a comment above this line explaining why we are checking for the `#`. Could easily see myself forgetting why the check is being done.
```suggestion
	// Check if name starts with # to detect non-element node names like #text and #comment
  const isInline = (name: string): boolean => !Strings.startsWith(name, '#') && isValid(name) && !isBlock(name);
```

Reply:
added